src/main/java/backend/Logic.java

```diff
@@ -312,6 +338,26 @@ private boolean handleIssueLabelsUpdateResult(boolean isUpdateSuccessful,
     }
 
     /**
+     * Handles the result of updating an issue's milestone on server.
+     * Locally reverts back to the original milestone if the server update fails.
+     * @param isUpdateSuccessful
+     * @param originalIssue
+     * @return true if the server update is successful
+     */
+    private boolean handleIssueMilestoneUpdateOnServerResult(boolean isUpdateSuccessful,
+                                                          Optional<TurboIssue> originalIssue) {
+        if (!originalIssue.isPresent()) {
+            logger.error("Unable to replace issue milestone locally");
+            return false;
+        }
+        if (isUpdateSuccessful) return true;
```

Comment:
This seems to be the lowest hanging fruit, so should be the first one to be picked?